mops/include/mops/checks.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef MOPS_CHECKS_HPP
+#define MOPS_CHECKS_HPP
+
+#include "mops/tensor.hpp"
+#include <algorithm>
+#include <cstddef>
+#include <cstdint>
+#include <stdexcept>
+#include <string>
+
+template <typename T_1, size_t N_DIMS_1, typename T_2, size_t N_DIMS_2>
```

Comment:
Shouldn't `T_1 == T_2` all the time? If so, we should call it `scalar_t` for consistency

Reply:
Unfortunately sometimes we're feeding integer tensors to this function. That's why they're two different template parameters

Reply:
ok, I would put that in the docs then!